### ISSUE TITLE
Add support to Cleanup commands

### ISSUE THREAD

**Issue by w0rk3r** [preludeorg/operator-support#399]
I believe this can be already on you guys roadmap, but it would be cool to see something like this in Operator.

Most use cases outside initial phases cause changes to the system that should be reverted, this would help in real envs as well as labs, and save a lot of time from manual cleanup.

**Comment by privateducky** [comment]
Yes, this is on our roadmap. No timeline yet. We followed this approach on CALDERA awhile back:
- TTPs get a new, optional cleanup property with a command reversing itself
- Logic added to the platform which can execute these clean ups in reverse order post-operation. 

We haven't added this functionality to Operator yet because we want to explore some alternative approaches, such as tying cleanup to the adversary (not TTP) level or creating cleanup payloads which are abstract from both TTP and adversary.

The main flaw in our CALDERA approach was how fragile the cleanup is on more complex operations, which include more fine-tuned cleanup or cleanup through lateral movement TTPs. 

More to come.

**Comment by w0rk3r** [comment]
Cool!
I understand that the approach of adversary level cleanup cover more use-cases, but it would be cool to have something in TTP level, at least for me, it is handy to test and write detections and also Operator TTPs (Using the deploy command) without having to rdp or reverse shell to make it manually. Maybe a deploy cleanup button? It would be awesome

**Comment by privateducky** [comment]
Yep, good to know. There's likely a multi-pronged approach for this - something at both TTP and adversary levels. 

We've been letting this one mull a bit vs rushing into it because we take TTP-format changes pretty carefully - so as to not back ourselves into a corner ;) 

If you have any thoughts or want to propose any TTP-level modifications, it'd be (more than) welcome. Just push a PR to the community repo with a TTP example and we can spark a discussion there.

**Comment by w0rk3r** [comment]
Sounds Awesome, David. I'll think a bit around this, then I'll open a PR for discussion, but I like the approach you guys used on Caldera.
